index.js

```diff
@@ -32,7 +32,11 @@ function keydown(event: KeyboardEvent): void {
 }
 
 function focusable(el: Focusable): boolean {
-  return el.tabIndex >= 0 && !el.disabled && !el.hidden && (!el.type || el.type !== 'hidden') && !el.closest('[hidden]')
+  return el.tabIndex >= 0 && !el.disabled && !el.hidden && (!el.type || el.type !== 'hidden') && visible(el)
+}
+
+function visible(el): boolean {
+  return el.offsetWidth > 0 || el.offsetHeight > 0
```

Comment:
Minor nit, but should `visible(el)` also contain the `!el.hidden` and `(!el.type || el.type !== 'hidden')` logic?